### DISCUSSION TITLE
Mutate data before saving to a pivot table

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#5719] [Help] [answered]
I have 3 tables: client, contact and client_contact, client_contact being a pivot table.
Alongside the ids of client and contact it should save additional data like the current user's id.

But I can't figure out how.

In the `ContactRelationManager` the `table` function looks like this:

```
public static function table(Table $table): Table
{
	return $table
		->columns([
			...
		])
		->filters([
			//
		])
		->headerActions([
			Tables\Actions\CreateAction::make()
				->mutateFormDataUsing(function (array $data): array {
					$data['user_id'] = auth()->id();
					return $data;
				})                

		])
	;
}   
```

I figured `mutateFormDataUsing` would do the trick but it doesn't. Is there another (similar) function for this purpose?

**Comment by danharrin** [comment, marked as answer]
Ensure that `user_id` is defined in the `withPivot()` method of the relationship
